cli/module_build.go

```diff
@@ -223,18 +249,79 @@ func (c *viamClient) printModuleBuildLogs(ctx context.Context, buildID, platform
 	return nil
 }
 
-func (c *viamClient) listModuleBuildJobs(moduleID moduleID, number *int32, buildIDFilter *string) (*buildpb.ListJobsResponse, error) {
+func (c *viamClient) listModuleBuildJobs(moduleIDFilter string, count *int32, buildIDFilter *string) (*buildpb.ListJobsResponse, error) {
 	if err := c.ensureLoggedIn(); err != nil {
 		return nil, err
 	}
 	req := buildpb.ListJobsRequest{
-		ModuleId:      moduleID.String(),
-		MaxJobsLength: number,
+		ModuleId:      moduleIDFilter,
+		MaxJobsLength: count,
 		BuildId:       buildIDFilter,
 	}
 	return c.buildClient.ListJobs(c.c.Context, &req)
 }
 
+// waitForBuildToFinish calls listModuleBuildJobs every moduleBuildPollingInterval
+// Will wait until the status of the specified job is DONE or FAILED
+// if platform is empty, it waits for all jobs associated with the ID.
+func (c *viamClient) waitForBuildToFinish(buildID, platform string) error {
```

Comment:
Question for @abe-winter : 

Should this have a timeout? This was written specifically for the action. Would you prefer to have timeout semantics defined using action tools or using a flag/field passed to the cli?

Reply:
Probably best... Something that prevents an endless poll when you forget about that one terminal window. We do have robots just constantly calling APIs so not too concerned.

Reply:
up to you. long term yes it should. for now, the primary user of these commands is the action which has its own timeout logic (as you said)

Reply:
> the primary user of these commands is the action which has its own timeout logic

If that is true, I think we should rely on that until we get a good understanding of what the timeout should be. Mostly because I'm not sure what a good default timeout would be. 2 hrs? A full setup+install could easily last 1hr. If anything, I almost wonder if the timeout should be server-side and not client side.... anyways.... thoughts for the future.

If it is possible to do it in the action, I think we should do it in the action. 

Otherwise people can do
`timeout 1hr viam module build logs --blah` (on linux)
and use
```
brew install coreutils
gtimeout 1hr viam module build logs --blah
```
on mac